File: exercises/advent2021/src/day7/part2.py
def parse(filename: str) -> list[int]:
    with open(filename) as file:
        line = file.read()
        return list(map(int, line.split(',')))

def costCalc(distance: int) -> int:
    absDist = abs(distance)
    return absDist*(absDist+1)//2

def solve(input: list[int]) -> int:
    input.sort()
    start = min(input)
    end = max(input)
    minCost = float('inf')

    for i in range(start, end+1):
        cost = sum([costCalc(crab - i) for crab in input])
        if cost < minCost:
            minCost = cost

    return minCost

def main():
    input = parse('data/day7.txt')
    result = solve(input)
    print(result)

if __name__ == '__main__':
    main()

File: exercises/advent-of-code/2021/clj/src/day1/part2.clj
(ns day1.part1
  (:require
   [day1.calc :as calc]
   [parser :as parser]))

(def input
  (parser/input parser/toInt "data/day1.txt"))

(def result
  (->> input
       (partition 3 1)
       (map #(apply + %))
       (calc/solve)))

(println result)